Automattic/wp-calypso
# Checkout: Move code that displays cart messages to wpcom-checkout package

#### Changes proposed in this Pull Request

During checkout (and other places that use the shopping cart), the shopping-cart endpoint may return messages that need to be displayed to the user. The component that performs this responsibility is `CartMessages`. In this PR we move the logic that parses the cart and displays the messages into the `wpcom-checkout` package so that this logic could be reused in other places that might use the shopping-cart package (eg: wp-admin).

It would be nice to move the entire `CartMessages` component, but the mechanism for rendering the messages depends on the API of the notices module in calypso, which cannot easily be externalized (yet). We could try to inject the notice action creators, but I don't think that would gain us much more than this current abstraction. 🤔 

#### Testing instructions

- Add a product to your cart and visit checkout.
- Click to enter a coupon and enter an invalid coupon code. Click the "Apply" button.
- Verify that you see an error message displayed in a dismissible dialog.

<img width="705" alt="Screen Shot 2021-03-22 at 1 43 07 PM" src="https://user-images.githubusercontent.com/2036909/112034203-9c21ce00-8b14-11eb-994f-f04251515723.png">
